nomad/resource_acl_auth_method.go

```diff
@@ -264,21 +279,54 @@ func resourceACLAuthMethodRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	log.Printf("[DEBUG] Reading ACL Auth Method %q", authMethodName)
-	authMethod, _, err := client.ACLAuthMethods().Get(authMethodName, nil)
+	authMethod, err := fetchResource(client, authMethodName)
 	if err != nil {
 		return fmt.Errorf("error reading ACL Auth Method %q: %s", authMethodName, err.Error())
 	}
 	log.Printf("[DEBUG] Read ACL Auth Method %q", authMethod.Name)
 
+	unredactResource(d, authMethod, nil)
+	setStateFromResource(d, authMethod)
+	return nil
+}
+
+func fetchResource(client *api.Client, name string) (*api.ACLAuthMethod, error) {
+	log.Printf("[DEBUG] Reading ACL Auth Method %q", name)
+	authMethod, _, err := client.ACLAuthMethods().Get(name, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error reading ACL Auth Method %q: %w", name, err)
+	}
+	log.Printf("[DEBUG] Read ACL Auth Method %q", name)
+	return authMethod, nil
+}
+
+func unredactResource(d *schema.ResourceData, fetchedAuthMethod, originalAuthMethod *api.ACLAuthMethod) {
```

Comment:
re:
> Note that it's impossible with or without this patch to import an Auth Method without manual intervention, because we can't create the OIDC client secret from nothing.

In my exp with Terraform, I wouldn't say this kind of thing is the norm, but it's not terribly uncommon to anticipate some extra hand-holding during a TF import.

I suppose the symptom would be that the very first `apply` after the import would show a diff, even if the secret in the user's HCL is actually identical to what Nomad has squirreled away in its state, yes?

Reply:
Yeah, because the TF state would be `"redacted"`